optaplanner-examples/src/test/java/org/optaplanner/examples/nurserostering/score/NurseRosteringConstraintProviderTest.java

```diff
@@ -37,31 +37,29 @@
 import org.optaplanner.examples.nurserostering.domain.request.ShiftOffRequest;
 import org.optaplanner.examples.nurserostering.domain.request.ShiftOnRequest;
 import org.optaplanner.test.api.score.stream.ConstraintVerifier;
+import org.optaplanner.test.api.score.stream.SingleConstraintVerification;
 
 class NurseRosteringConstraintProviderTest
         extends AbstractConstraintProviderTest<NurseRosteringConstraintProvider, NurseRoster> {
 
-    private final AtomicLong idSupplier = new AtomicLong(0);
-    private final Map<Pair<Integer, ShiftType>, Shift> indexShiftTypePairToShiftMap = new ConcurrentHashMap<>();
-    private final Map<Integer, ShiftDate> indexToShiftDateMap = new ConcurrentHashMap<>();
+    private final ThreadLocal<AtomicLong> idSupplier = ThreadLocal.withInitial(() -> new AtomicLong(0));
+    private final ThreadLocal<Map<Pair<Integer, ShiftType>, Shift>> indexShiftTypePairToShiftMap =
+            ThreadLocal.withInitial(HashMap::new);
+    private final ThreadLocal<Map<Integer, ShiftDate>> indexToShiftDateMap = ThreadLocal.withInitial(HashMap::new);
```

Comment:
This way the bug I introduced.
Originally I added ConcurrentHashMap, thinking that the issue I was seeing came from the fact that multiple threads where concurrently modifying the map.
Only at closer inspection now have I realized that multiple threads must not share these maps at all.